api/Makefile

```diff
@@ -2,88 +2,105 @@ include .env.development*
+	@echo "Setting up tools..."
+	@test -x $(shell go env GOPATH)/bin/golangci-lint || \
+		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.41.1/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.41.1
 
+.PHONY: fmt
 fmt:
 	@echo "Formatting code..."
 	gofmt -s -w ${SRC_ROOT}
 
-dep-ci:
-	@echo "Getting CI dependencies..."
-	@test -x ${GOPATH}/bin/golangci-lint || go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.27.0
-
-lint:
-	@echo "Linting code..."
-	golangci-lint -v run --timeout 3m
-
-lint-fix:
+.PHONY: lint
+lint: setup
 	@echo "Linting code..."
-	golangci-lint -v run --timeout 3m --fix
-
-lint-ci: dep-ci lint
+	golangci-lint -v run --timeout 3m $(if $(filter true,$(fix)),--fix,)
 
+.PHONY: tidy
 tidy:
 	@echo "Fetching dependencies..."
 	go mod tidy
 
-vendor: setup
+.PHONY: vendor
+vendor:
 	@echo "Fetching dependencies..."
 	go mod vendor
 
-test: tidy local-db
+.PHONY: test
+test: tidy
 	@echo "Running tests..."
 	go test -v -race -short -cover -coverprofile cover.out ${SRC_ROOT}/... -tags unit,integration
 	go tool cover -func cover.out
 
-test-ci: tidy
-	echo "Running tests..."
-	DATABASE_HOST=$${POSTGRES_HOST} \
-	DATABASE_USER=$${POSTGRES_USER} \
-	DATABASE_PASSWORD=$${POSTGRES_PASSWORD} \
-	DATABASE_NAME=$${POSTGRES_DB} \
-	go test -v -race -short -cover -coverprofile cover.out ${SRC_ROOT}/... -tags unit,integration
-	go tool cover -func cover.out
-
-test-e2e-local: deploy_docker_stack
-	@$(eval include ./e2e/local/.env.testing)
-	@echo "Running e2e tests..."
-	go clean -testcache
-	API_BASE_PATH=${E2E_API_BASE_PATH} \
-		TEST_ID=$(if $(TEST_ID),$(TEST_ID),$(shell date +%Y%m%d%H%M)) \
-		go test -v -parallel=2 ${E2E_TEST_ROOT}/... -tags e2e -timeout 15m -run TestEndToEnd || true
-	make clean_docker_stack
-
+.PHONY: build
 build: tidy
 	@echo "Building binary..."
 	go build -o ./bin/${BIN_NAME} ${SRC_ROOT}/cmd/main.go
 
+.PHONY: build-image
+build-image: vendor version
+	@$(eval IMAGE_TAG = $(if $(DOCKER_REGISTRY),$(DOCKER_REGISTRY)/,)${BIN_NAME}:${VERSION})
+	@echo "Building docker image: ${IMAGE_TAG}"
+	docker build --tag ${IMAGE_TAG} .
+
+.PHONY: run
 run: build local-db
 	./bin/${BIN_NAME} -config config-dev.yaml
 
+.PHONY: local-db
 local-db:
 	@docker-compose up -d postgres && docker-compose run migrations
 
+.PHONY: swagger-ui
 swagger-ui:
 	@docker-compose up -d swagger-ui
 	@xdg-open 2>/dev/null http://localhost:8081 || open http://localhost:8081
 
+.PHONY: gen-client
+gen-client:
+	rm -rf turing/generated
+	@docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli:v5.1.1 generate \
+		--input-spec /local/api/openapi-sdk.yaml \
+		--generator-name go \
+		--output /local/turing/generated \
+		--config /local/openapi-codegen.yaml
+
+.PHONY: version
+version:
+	$(eval VERSION=$(if $(OVERWRITE_VERSION),$(OVERWRITE_VERSION),$(shell ../scripts/vertagen/vertagen.sh -f docker)))
+	@echo "turing-api version:" $$VERSION
+
+
+#   R E V I E W   R E Q U I R E D
```

Comment:
It means that I doubt that it was ever tested locally and I have a reason to suspect that it's broken and won't work as expected. I decided to keep it in the repo for now for the reference, but this should be reviewed and updated. 

> Will this be tackled in a future PR?

Not sure if it will come as a part of my future PR. Will add it as a card to our backlog